contracts/Pool.sol

```diff
@@ -1,31 +1,292 @@
 // SPDX-License-Identifier: GPL-3.0-only
 pragma solidity ^0.6.2;
 
+/// @notice Donation pool contract.
+/// Allows sending ETH automatically to a configurable set of receivers.
 contract Pool {
+    using ReceiverWeightsImpl for mapping(address => ReceiverWeight);
+
+    /// @notice On every block `B`, which is a multiple of `cycleBlocks`, the receivers
+    /// gain access to funds collected from all blocks from `B - cycleBlocks` to `B - 1`.
+    uint64 internal immutable cycleBlocks;
+    /// @notice Block number on which all payments must be finished
+    uint64 internal immutable maxBlockNumber;
+    /// @notice Maximum sum of all receiver weights of a single sender.
+    /// Limits loss of per-block payments accuracy, they are always multiples of weights sum.
+    uint32 internal constant SENDER_WEIGHTS_SUM_MAX = 1000;
+    /// @notice Maximum number of receivers of a single sender.
+    /// Limits costs of changes in sender's configuration.
+    uint32 internal constant SENDER_WEIGHTS_COUNT_MAX = 100;
+
     struct Sender {
-        uint256 balance;
+        /// @notice Block number on which funding has been started
+        uint64 startBlock;
+        /// @notice The amount available when funding has been started
+        uint192 startBalance;
+        // --- SLOT BOUNDARY
+        /// @notice The target amount sent on each block.
+        /// The actual amount is rounded down to the closes multiple of `weightSum`.
+        uint192 amtPerBlock;
+        /// @notice The total weight of all the receivers
+        uint32 weightSum;
+        /// @notice The number of the receivers
+        uint32 weightCount;
+        // --- SLOT BOUNDARY
+        /// @notice The mapping of all the receivers to their weights, iterable
+        mapping(address => ReceiverWeight) receiverWeights;
+    }
+
+    struct Receiver {
+        /// @notice The block until which funds have been collected previously
+        uint64 lastCollectedCycle;
+        /// @notice The amount which has been previously collected
+        uint192 lastCollectedAmt;
```

Comment:
This tripped my up because it’s not really the amount that was last collected (potentially for multiple cycles) but the per cycle amount when we last calculated it. How about this:
```suggestion
        /// @notice The amount of funds received for `lastCollectedCycle`.
        uint192 lastFundsPerCycle;
```

Reply:
Fixed